tests/foreman/cli/test_contentview.py

```diff
@@ -1012,6 +1013,89 @@ def test_positive_add_custom_repo_by_name(self):
+
+        @Assert: Module was added and has latest version.
+
+        @CaseLevel: Integration
+        """
+        module = {'name': 'versioned', 'version': '3.3.3'}
+        puppet_repository = make_repository({
+            u'content-type': u'puppet',
+            u'product-id': self.product['id'],
+            u'url': CUSTOM_PUPPET_REPO,
+        })
+        puppet_module = PuppetModule.list({
+            'search': 'name={0} and version={1}'.format(
+                module['name'], module['version'])
+        })[0]
+        Repository.synchronize({'id': puppet_repository['id']})
+        content_view = make_content_view({u'organization-id': self.org['id']})
+        ContentView.puppet_module_add({
+            u'content-view-id': content_view['id'],
+            u'name': puppet_module['name'],
+            u'author': puppet_module['author'],
+        })
+        cv_module = ContentView.puppet_module_list(
+            {'content-view-id': content_view['id']})
+        self.assertGreater(len(cv_module), 0)
+        self.assertIn(puppet_module['version'], cv_module[0]['version'])
+        self.assertIn('Latest', cv_module[0]['version'])
+
+    @tier2
+    @run_only_on('sat')
+    def test_positive_add_puppet_module_older_version(self):
+        """Add older version of puppet module to Content View by id/uuid
+
+        @id: 39654b3e-963f-4859-81f2-9992b60433c2
+
+        @Steps:
+
+        1. Upload/sync puppet repo with several versions of the same module
+        2. Add older version by id/uuid to CV
+
+        @Assert: Exact version (and not latest) was added.
+
+        @CaseLevel: Integration
+
+        @BZ: 1240491
+        """
+        module = {'name': 'versioned', 'version': '2.2.2'}
+        puppet_repository = make_repository({
+            u'content-type': u'puppet',
+            u'product-id': self.product['id'],
+            u'url': CUSTOM_PUPPET_REPO,
+        })
+        Repository.synchronize({'id': puppet_repository['id']})
+        puppet_module = PuppetModule.list({
+            'search': 'name={0} and version={1}'.format(
+                module['name'], module['version'])
+        })[0]
+        for by in ('uuid', 'id'):
+            with self.subTest(by=by):
+                content_view = make_content_view(
+                    {u'organization-id': self.org['id']})
+                ContentView.puppet_module_add({
+                    u'content-view-id': content_view['id'],
+                    by: puppet_module[by],
+                })
+                cv_module = ContentView.puppet_module_list(
+                    {'content-view-id': content_view['id']})
+                self.assertGreater(len(cv_module), 0)
+                self.assertEqual(cv_module[0]['version'], module['version'])
+                # Remove puppet module before next iteration
+                ContentView.puppet_module_remove({
+                    'content-view-id': content_view['id'],
+                    'uuid': puppet_module['uuid']
```

Comment:
I thought about that but looks like `content-view puppet-module add` and `content-view puppet-module remove` use different ids. As it it possible to add puppet module by its global id, but not to remove. Or perhaps it is a bug. I'm not sure hot to check that as `content-view puppet-module remove`  does not display any ids except uuid.

```
hammer> puppet-module list
---|--------------------------|------------|---------|-------------------------------------
ID | NAME                     | AUTHOR     | VERSION | UUID                                
---|--------------------------|------------|---------|-------------------------------------
11 | ui_test_variables        | robottelo  | 0.2.0   | daa047a1-3e3a-425c-bb09-659474619ac3

hammer> content-view puppet-module add --content-view-id 66 --id 11
Puppet module added to content view

hammer> content-view puppet-module remove --content-view-id 66 --id 11
Couldn't remove puppet module from the content view:
  Error: content_view_puppet_module not found

hammer> content-view puppet-module list --content-view-id 66       
-------------------------------------|-------------------|-----------|--------
UUID                                 | NAME              | AUTHOR    | VERSION
-------------------------------------|-------------------|-----------|--------
daa047a1-3e3a-425c-bb09-659474619ac3 | ui_test_variables | robottelo | 0.2.0  
```